src/test/stub/len_vec.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+extern crate libc;
+
+use std::mem;
+use std::ptr::{write, read};
+
+// Abstraction which tracks only the length and the last element of the vector.
+pub struct Vec<T> {
+    len: usize,
+    last: Option<*mut T>,
```

Comment:
Why do we track the last element?

Reply:
Do we have an example of a proof where this helps

Reply:
What if we track the `ith` element? And the first time we try to get that element, we assume i was the requested index?

Reply:
```
for i in 1..10 {
    v[i] = i;
}

assert!(v[3] == 3);
```